README.md

```diff
@@ -53,7 +52,21 @@ class ExpensiveRenderer extends React.Component {
 }
 ```
 
-## Benchmarking
+## Do I Need `shouldComponentUpdate`?
+
+> What's faster than a really fast deep comparion? No deep comparison at all.
+
+—This Readme
+
+
+Deep checks in React's `shouldComponentUpdate` should not be used blindly.
+First, see if a
+[PureComponent](https://reactjs.org/docs/react-api.html#reactpurecomponent) would
+work for you. If it won't (if you need deep checks), make sure you've correctly indentified the bottleneck in
+your application by
+[profiling the performance](https://reactjs.org/docs/optimizing-performance.html#profiling-components-with-the-chrome-performance-tab). For more information about making your app faster, check out the [Optimizing Performance](https://reactjs.org/docs/optimizing-performance.html) section of the React docs.
```

Comment:
I'd add a final line to this like: `After you've assessed that you _do_ indeed need deep equality checks and identified the minimum number of places to apply them, then this library may be for you!`